test/test_mongo_middleware.js

```diff
@@ -238,6 +240,48 @@ describe('mongo db middleware', function() {
       });
     });
 
+    it('passes isForSubmit = true in options when fields has $submit = true', function(done) {
+      sinon.spy(Collection.prototype, 'find');
+      var middlewareSpy = sinon.spy(function(request, next) {
+        expect(request).to.have.all.keys([
+          'action',
+          'collectionName',
+          'options',
+          'query'
+        ]);
+        expect(request.action).to.equal(BEFORE_SNAPSHOT_LOOKUP);
+        expect(request.collectionName).to.equal('testcollection');
+        expect(request.options.testOptions).to.equal('yes');
+        expect(request.options.isForSubmit).to.equal(true);
+        expect(request.query._id).to.equal('test1');
+        // test that when we set queryOptions in the middleware, they get passed to the Mongo driver.
+        request.queryOptions = {
+          isForSubmit: true
```

Comment:
I'd use a real option to `Collection#find`, like `maxTimeMS`. We do want to TypeScript-ify this one day, and using a real option will make the typing a bit easier.